Sources/FileStream.swift

```diff
@@ -43,37 +43,41 @@ public final class FileStream: Stream {
         self.highWaterMark = highWaterMark
     }
 
-    public func receive() throws -> Data {
+    public func receive(upTo byteCount: Int, timingOut deadline: Double = .never) throws -> Data {
         try assertNotClosed()
         do {
             if file.eof {
                 file.close()
                 return []
             } else {
-                return try file.read(lowWaterMark: lowWaterMark, highWaterMark: highWaterMark)
+                return try file.read(lowWaterMark: lowWaterMark, highWaterMark: byteCount, timingOut: deadline)
```

Comment:
Yeah, that should work. So the user will be always sure that nothing will overcome `highWaterMark`.